inst/tests/tests.Rraw

```diff
@@ -3729,6 +3729,15 @@ test(1100, dt1[dt2,roll=-Inf,rollends=c(FALSE,TRUE)]$ind, INT(NA,NA,1,2,2,2,2,2,
   DT = data.table(x=sample(5,20,TRUE), y=sample(2,20,TRUE), z=sample(letters[1:2],20,TRUE), d1=runif(20), d2=1L)
   test(1102.38, names(dcast(DT, x ~ y + z, fun.aggregate=length, value.var = "d2", sep=".")),
                 c("x", "1.a", "1.b", "2.a", "2.b"))
+
+  # test for #5512, only compute default fill if needed.
+  DT = data.table(chr=c("a","b","b"), int=1:3, dbl=as.double(4:6))
+  test(1102.39, dcast(DT, int ~ chr, min, value.var="int"), data.table(int=1:3, a=c(1L,NA,NA), b=c(NA,2L,3L), key="int"), warning="NAs introduced by coercion to integer range")#as.integer(min(integer()) is Inf) is NA
+  test(1102.391, dcast(DT, int ~ chr, min, value.var="int", fill=NA), data.table(int=1:3, a=c(1L,NA,NA), b=c(NA,2L,3L), key="int"))
```

Comment:
this is the same output as the previous test, right? if so, consider doing `ans<-data.table(...)` in the previous test and just `ans` here. I think that will make the test series more readable as we don't need to mentally construct a new table when checking our intuition for how the test should work.

Reply:
ok